### PR TITLE
Handle error model

### PR DESCRIPTION
**Pull request by AhmedAtef277**

### CODE REVIEW

File: network/Sources/network/ApiError.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+//
+//  File.swift
+//  
+//
+//  Created by Ahmed Atef on 23/03/2023.
+//
+
+import Foundation
+
+public enum ApiError: Error {
```
Comment (Mahmoud-Bassuni) [inline network/Sources/network/ApiError.swift:10]:
```suggestion
public enum RemoteError: Error {
```

File: network/Sources/network/ResponseHandler.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+//  
+//
+//  Created by Ahmed Atef on 09/04/2023.
+//
+
+import Foundation
+import Alamofire
+
+class ResponseHandler<T: Codable> {
+    static func handle(response: AFDataResponse<String>, route: ServiceLayer, completion: @escaping (Result<T, ApiError>) -> Void) {
```
Comment (Mahmoud-Bassuni) [inline network/Sources/network/ResponseHandler.swift:12]:
```suggestion
    static func responseData(response: AFDataResponse<String>, route: ServiceLayer, completion: @escaping (Result<T, ApiError>) -> Void) {
```

File: network/Sources/network/ApiError.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    case detectError(statusCode: Int)
+    var description: String {
+        switch self {
+        case .detectError(let statusCode):
+            switch statusCode {
+            case 100...199: return "Server has received the request and is continuing to process it."
```
Comment (Mahmoud-Bassuni) [inline network/Sources/network/ApiError.swift:16]: try to fine more descriptive message 😀

Comment (AhmedAtef277) [reply network/Sources/network/ApiError.swift:16]: any suggestion?😅

**Review by Mahmoud-Bassuni — APPROVED** [review]

**Review by hassanSayed0 — APPROVED** [review]

**Pull request merged**